src/nyc_trees/apps/survey/layer_context.py

```diff
@@ -83,6 +83,10 @@ def get_context_for_territory_admin_layer(group_id):
                                   {'group': group_id})
 
 
+def get_group_territory_modification_time(request):
```

Comment:
This doesn't seem to account for scenarios where groups get deleted, or when borderers are edited. Since the request is quick after removing the `JOIN`, it seems fine to leave it as is without introducing `Last-Modified`.